### ISSUE TITLE
Request Body Internal Ref Not Resolved

### ISSUE THREAD

**Issue by nkringle** [swagger-api/swagger-parser#855]
I'm not sure if this is a duplicate/related to #764 or #731. 

Basically, when OpenApiV3Parser parses a post request body with a schema ref, the ref isn't being resolved for the request body.

Here's the openapi v3:
```yaml
openapi: '3.0.0'
info:
  title: Pet Store API
  version: '3.0'

paths:
  /adopt:
    post:
      requestBody:
        $ref: '#/components/requestBodies/ASinglePet'
      responses:
        201:
          $ref: '#/components/responses/Adopted'

components:
  requestBodies:
    ASinglePet:
      description: Adoption Request
      required: true
      content:
        application/petstore+json:
          schema:
            $ref: '#/components/schemas/AdoptionRequest'
  responses:
    Adopted:
      description: Pet successfully adopted
      content:
        application/petstore+json:
          schema:
            $ref: '#/components/schemas/AdoptionResponse'

  schemas:
    Pets:
      type: object
      properties:
        pets:
          type: array
          items:
            $ref: '#/components/schemas/Pet'
    Pet:
      type: object
      properties:
        name:
          type: string
          example: Fluffy
    AdoptionRequest:
      description:  Request to adopt a new pet
      type: object
      required:
      - name
      - pets
      properties:
        name:
          description: Name pet store
          type: string
        pets:
          $ref: '#/components/schemas/Pets'
    AdoptionResponse:
      description: Adoption Response
      type: object
      properties:
        string:
          description: Uri for the adopted pet
          type: string
```

Here's the request (formatted as a test 😉):
```java
     ParseOptions parseOptions = new ParseOptions();
        parseOptions.setResolve(true);
        OpenAPI openAPI = new OpenAPIV3Parser().read("src/test/resources/openApiSpec.yaml", null, parseOptions);

        Content actualComponentContent = openAPI.getComponents().getRequestBodies().get("ASinglePet").getContent();
        Content actualPathContent = openAPI.getPaths().get("/adopt").getPost().getRequestBody().getContent();
        Map properties = actualComponentContent.get("application/petstore+json").getSchema().getProperties();

        assertNotNull(properties);
        assertEquals(properties.size(), 2);

        assertNotNull(actualPathContent);
        assertEquals(actualPathContent, actualComponentContent);
```

I'd expect the schema content to be populated under the path: `/adopt/post/reqeustBody/content`, but instead it's only the `ref` populated. 

If you want me to submit a PR for this I'd be more than happy. (I'm also not ruling out that I'm using ParseOptions, etc. correctly). 😄

**Comment by gracekarina** [comment]
Hi! @nkringle, thanks for spotting this out. Please send us a PR with a test so we can look at it and find a solution.

**Comment by nkringle** [comment]
Will do, thanks!

**Comment by ymohdriz** [comment]
Hi @nkringle,

Please use the below code:

        OpenAPIParser openApiParser = new OpenAPIParser();
        ParseOptions options = new ParseOptions();
        options.setResolve(true);
        options.setFlatten(true);
        OpenAPI openAPI = openApiParser.readLocation("issue799.json", null, options).getOpenAPI();

setFlatten(true) will replace the $ref with the actual component.

Thanks,
Mohammed

**Comment by nkringle** [comment]
@ymohdriz making this change (since I'm using V3), did not change the outcome/make my test pass. 
```
 OpenAPIV3Parser openAPIV3Parser = new OpenAPIV3Parser();
ParseOptions parseOptions = new ParseOptions();
parseOptions.setResolve(true);
parseOptions.setFlatten(true);

OpenAPI openAPI = openAPIV3Parser.readLocation("src/test/resources/issue_855.yaml", null, parseOptions).getOpenAPI();

```

**Comment by ymohdriz** [comment]
Hi @nkringle,

Sorry for the confusion. setResolveFully(true) should solve the issue. It's looks like a bug. I have fixed it and request a PR #858 .

Thanks,
Mohammed

**Comment by gracekarina** [comment]
Closed fix by #858
